src/soft-trigger.c

```diff
@@ -35,7 +51,8 @@ SR_PRIV struct soft_trigger_logic *soft_trigger_logic_new(
 	stl = g_malloc0(sizeof(struct soft_trigger_logic));
 	stl->sdi = sdi;
 	stl->trigger = trigger;
-	stl->unitsize = (g_slist_length(sdi->channels) + 7) / 8;
+	/* Retreive number of logic channels, unitsize */
+	stl->unitsize = logic_channel_unitsize(sdi->channels);
```

Comment:
What's wrong with the implementation that was there before? Why is logic_channel_unitsize() needed?

Reply:
sdi->channels contains analog and logic channels. number of channels and unitsize can than be wrong.